File: programmers/js/level00/120892.js
// 암호 해독
function solution(cipher, code) {
  return cipher
    .split('')
    .map((val, idx) => (idx + 1) % code === 0 ? val : null)
    .join('');
}
// 다른 풀이
// function solution(cipher, code) {
//   return cipher.split('').filter((_, index) => (index + 1) % code === 0).join('');
// }

File: programmers/java/level00/120892.java
// 암호 해독

import java.util.stream.IntStream;
import java.util.stream.Collectors;

class Solution {
    public String solution(String cipher, int code) {
        return IntStream.range(0, cipher.length())
            .filter(idx -> (idx + 1) % code == 0)
            .mapToObj(idx -> String.valueOf(cipher.charAt(idx)))
            .collect(Collectors.joining());
    }
}